### PR TITLE
Logging Policy During Optimization

### PR DESCRIPTION
**Pull request by justinkterry**
## Description
 Closes issue #113. I haven't fully tested this yet; once this looks appropriately shaped to you I will.
  
  ## Types of changes
  
  - [ ] Bug fix (non-breaking change which fixes an issue)
  - [x] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to change)
  - [ ] Documentation (update in the documentation)
  
  ## Checklist:
  
  
  - [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
  - [x] I have updated the changelog accordingly (**required**).
  - [ ] My change requires a change to the documentation.
  - [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
  - [ ] I have updated the documentation accordingly.
  - [x] I have reformatted the code using `make format` (**required**)
  - [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
  - [x] I have ensured `make pytest` and `make type` both pass. (**required**)

### CODE REVIEW

**Comment by araffin** [comment]
Hello,
thanks for the PR.
please keep the PR template.

**Comment by justinkterry** [comment]
Tests pass and the template is filled out

File: train.py [hunk 1]
```diff
@@ -40,6 +40,12 @@
         default=10000,
         type=int,
     )
+    parser.add_argument(
+        "--logging",
```
Comment (araffin) [inline train.py:44]: maybe `optimization-log-path` (or something similar) to be more explicit and avoid confusion vs normal training?

File: utils/callbacks.py [hunk 2]
```diff
@@ -26,6 +26,8 @@ def __init__(
         eval_freq: int = 10000,
         deterministic: bool = True,
         verbose: int = 0,
+        best_model_save_path: str = None,
```
Comment (araffin) [inline utils/callbacks.py:29]: type should be `Optional[str]`, no?

File: utils/exp_manager.py [hunk 3]
```diff
@@ -607,13 +609,24 @@ def objective(self, trial: optuna.Trial) -> float:
...
         optuna_eval_freq = max(optuna_eval_freq // model.get_env().num_envs, 1)
         # Use non-deterministic eval for Atari
-        eval_callback = TrialEvalCallback(
-            eval_env,
-            trial,
-            n_eval_episodes=self.n_eval_episodes,
-            eval_freq=optuna_eval_freq,
-            deterministic=self.deterministic_eval,
-        )
+        if self.logging != "":
```
Comment (araffin) [inline utils/exp_manager.py:612]: please use the if/else only for defining the variables that change (clearer to see the difference) so you have only one call to the constructor.

**Review by araffin — COMMENTED** [review]
The PR almost looks good (only minor comments).
As mentioned in the PR template (required for new features), please also add a small integration test too (that checks that the log files are correctly created).

**Comment by justinkterry** [comment]
Comments are fixed and a working test that passes has been added

**Review by araffin — APPROVED** [review]
LGTM, thanks =)

**Pull request merged**